server/src/main/java/keywhiz/service/daos/SecretSeriesDAO.java

```diff
@@ -238,17 +237,21 @@ public void deleteSecretSeriesByName(String name) {
   public void deleteSecretSeriesById(long id) {
     long now = OffsetDateTime.now().toEpochSecond();
     dslContext.transaction(configuration -> {
-      DSL.using(configuration)
-          .update(SECRETS)
-          .set(SECRETS.CURRENT, (Long) null)
-          .set(SECRETS.UPDATEDAT, now)
-          .where(SECRETS.ID.eq(id))
-          .execute();
-
-      DSL.using(configuration)
-          .delete(ACCESSGRANTS)
-          .where(ACCESSGRANTS.SECRETID.eq(id))
-          .execute();
+      SecretsRecord r = DSL.using(configuration)
+          .fetchOne(SECRETS, SECRETS.ID.eq(id).and(SECRETS.CURRENT.isNotNull()));
```

Comment:
I think this should maybe call `.forUpdate()`, so the row is locked until the subsequent UPPDATE

Reply:
Updated to use forUpdate on this select.